src/mappings/mappingHandlers.ts

```diff
@@ -1,23 +1,79 @@
-import { SubstrateBlock, SubstrateExtrinsic, SubstrateEvent } from '@subql/types'
-import { BlockHandler } from '../handlers/block'
-import { EventHandler, ExtrinsicHandler } from '../handlers'
-import { CallHandler } from '../handlers/call'
-import { VestingScheduleHandler } from '../handlers/vestingschedule'
+import { EventRecord } from "@polkadot/types/interfaces";
+import { SubstrateExtrinsic, SubstrateBlock } from "@subql/types";
+import { SpecVersion, Event, Extrinsic } from "../types";
 
+let specVersion: SpecVersion;
 export async function handleBlock(block: SubstrateBlock): Promise<void> {
-    const handler = new BlockHandler(block)
+  // Initialise Spec Version
+  if (!specVersion) {
+    specVersion = await SpecVersion.get(block.specVersion.toString());
+  }
 
-    await handler.save()
-}
+  // Check for updates to Spec Version
+  if (!specVersion || specVersion.id !== block.specVersion.toString()) {
+    specVersion = new SpecVersion(block.specVersion.toString());
+    specVersion.blockHeight = block.block.header.number.toBigInt();
+    await specVersion.save();
+  }
```

Comment:
```suggestion
  if (!specVersion.blockHeight) {
    specVersion.blockHeight = block.block.header.number.toNumber();
    await specVersion.save();
  }
```

Reply:
The suggestion is above based on another suggestion to use Int instead of BigInt for block numbers